toqito/state_opt/tests/test_ppt_distinguishability.py

```diff
@@ -1,215 +1,112 @@
+    # Primal result - min error
+    (states_YYD, probs_YYD, "min-error", True, 7/8),
+    # Dual Result - min error
+    (states_YYD, probs_YYD, "min-error", False, 7/8),
+    # Primal result - unambiguous
+    (states_YYD, probs_YYD, "unambiguous", True, 3/4),
+    # Dual Result - unambiguous
+    (states_YYD, probs_YYD, "unambiguous", False, 3/4),
+    # PPT distinguishing the YYD states from :cite:`Yu_2012_Four` should yield `7/8 ~ 0.875`.
+    # Feeding the input to the function as vectors.
+    # Primal result - min error
+    (vec_YYD, probs_YYD, "min-error", True, 7/8),
+    # Dual Result - min error
+    (vec_YYD, probs_YYD, "min-error", False, 7/8),
+    # Primal result - unambiguous
+    (vec_YYD, probs_YYD, "unambiguous", True, 3/4),
+    # Dual Result - unambiguous
+    (vec_YYD, probs_YYD, "unambiguous", False, 3/4),
+    # PPT distinguishing the YYD states from :cite:`Yu_2012_Four` should yield `7/8 ~ 0.875`.
+    # Feeding the input to the function as density matrices.
+    # Primal result - min error without probs
+    (states_YYD, None, "min-error", True, 7/8),
+    # Dual Result - min error without probs
+    (states_YYD, None, "min-error", False, 7/8),
+    # Primal result - unambiguous without probs
+    (states_YYD, None, "unambiguous", True, 3/4),
+    # Dual Result - unambiguous without probs
+    (states_YYD, None, "unambiguous", False, 3/4),
+    # One quantum data hiding scheme involves the Werner hiding pair :cite:`Terhal_2001_Hiding`. A Werner hiding pair
+    # is defined by :cite:`Cosentino_2015_QuantumState`.
+    # Primal result - min error without probs
+    (states_werner, None, "min-error", True, expected_val_werner),
+    # Dual Result - min error without probs
+    (states_werner, None, "min-error", False, expected_val_werner),
+    # Primal result - unambiguous without probs
+    (states_werner, None, "unambiguous", True, 1/3),
+    # Dual Result - unambiguous without probs
+    (states_werner, None, "unambiguous", False, 1/3),
+    # PPT distinguishing the four Bell states. There exists a closed form formula for the probability with which one
+    # is able to distinguish one of the four Bell states given with equal
+    # probability when Alice and Bob have access to a resource state :cite:`Bandyopadhyay_2015_Limitations`.
+    # Refer to Theorem 5 in  :cite:`Bandyopadhyay_2015_Limitations` for more details.
+    # Primal result - min error
+    (states_bell_resource, probs_bell, "min-error", True, exp_res_bell),
+    # Dual result - min error
+    (states_bell_resource, probs_bell, "min-error", False, exp_res_bell)])
+def test_ppt_distinguishability(input_states, input_probs, dist_method_str, strategy_bool, expected_result):
+    """Tests function works as expected for valid inputs."""
+    calculated_result = ppt_distinguishability(
+        input_states, probs=input_probs, dist_method=dist_method_str, strategy=strategy_bool)
+    assert np.isclose(calculated_result, expected_result, atol = 0.001)
```

Comment:
All of the `min-error` methods used in the tests here are for the `dual problem`. We want to introduce a test that does the same thing for the `primal problem`. 

Maybe the labels for `primal_res` and `dual_res` were not used correctly in existing tests?

Reply:
Yes, definitely. For this SDP, strong duality holds so we should expect the primal and dual values to be the same. Definitely something we should verify via testing, so good point!

Reply:
Sure, I do understand that both values should be the same. 

I am a bit confused as to why code coverage is flagging the following lines from `primal_problem` as uncovered considering there are multiple tests for these lines. 

![image](https://github.com/vprusso/toqito/assets/66048318/ad77f566-adbe-4a23-a422-4e60f65193bd)

`if-else` conditions from the main function:

![image](https://github.com/vprusso/toqito/assets/66048318/b96fbff1-a749-4d89-a7d1-9356eab3af19)

The test shown below should have covered the flagged lines:

![image](https://github.com/vprusso/toqito/assets/66048318/674803d3-4f31-4bf0-95e4-0372f222b572)

Reply:
Interestingly, in the codecov report, there are no flagged lines. 

![image](https://github.com/vprusso/toqito/assets/66048318/332f8c34-a432-42b4-9ae8-da7248300d7e)

But in the checks we run, `pytest` flags those lines as uncovered. 

![image](https://github.com/vprusso/toqito/assets/66048318/b4c18ab0-824f-4fc5-b723-2e91cacd3b9a)